assets/js/modules/analytics-4/components/widgets/TopCitiesWidget.js

```diff
@@ -21,15 +21,97 @@
  */
 import PropTypes from 'prop-types';
 
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
+import { ZeroDataMessage } from '../../../analytics/components/common';
+import { numFmt } from '../../../../util';
+import { MetricTileTable } from '../../../../components/KeyMetrics';
+const { useSelect, useInViewSelect } = Data;
+
 export default function TopCitiesWidget( { Widget } ) {
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} )
+	);
+
+	const topcCitiesReportOptions = {
+		...dates,
+		dimensions: [ 'city' ],
+		metrics: [ { name: 'totalUsers' } ],
+		orderby: [
+			{
+				metric: {
+					metricName: 'totalUsers',
+				},
+				desc: true,
+			},
+		],
+		limit: 3,
+	};
+
+	const topCitiesReport = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( topcCitiesReportOptions )
+	);
+
+	const loading = useInViewSelect(
+		( select ) =>
+			! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+				'getReport',
+				[ topcCitiesReportOptions ]
+			)
+	);
+
+	const { rows = [] } = topCitiesReport || {};
+
+	const totalUsers =
+		topCitiesReport?.totals?.[ 0 ]?.metricValues?.[ 0 ]?.value;
```

Comment:
We're already destructuring from topCitiesReport above with a fallback, how about we do the same with `totals` above as well? We'd still need some chaining here but it could be simplified a bit.